src/main/java/teammates/ui/webapi/GetHasResponsesAction.java

```diff
@@ -87,7 +101,20 @@ JsonResult execute() {
             return new JsonResult(new HasResponsesData(hasResponses));
         } else {
             String courseId = getNonNullRequestParamValue(Const.ParamsNames.COURSE_ID);
-            String feedbackSessionName = getNonNullRequestParamValue(Const.ParamsNames.FEEDBACK_SESSION_NAME);
+            String feedbackSessionName = getRequestParamValue(Const.ParamsNames.FEEDBACK_SESSION_NAME);
+            if (feedbackSessionName == null) {
+                // check all sessions in the course
+                List<FeedbackSessionAttributes> feedbackSessions = logic.getFeedbackSessionsForCourse(courseId);
+                StudentAttributes student = logic.getStudentForGoogleId(courseId, userInfo.getId());
+
+                Map<String, Boolean> sessionsHasResponses = new HashMap<>();
+                for (FeedbackSessionAttributes feedbackSession : feedbackSessions) {
+                    boolean hasResponses = logic.hasStudentSubmittedFeedback(feedbackSession, student.email);
```

Comment:
maybe we can inline this value inside the method below

Reply:
Will it be too verbose in one line? Separating it this way could still work at no additional cost and improved readability.

Reply:
hmm yes I agree, lets leave it this way